src/main/java/io/jenkins/plugins/services/impl/HttpClientJiraIssues.java

```diff
@@ -0,0 +1,98 @@
+package io.jenkins.plugins.services.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import io.jenkins.plugins.models.JiraIssue;
+import io.jenkins.plugins.models.JiraIssues;
+import io.jenkins.plugins.services.ConfigurationService;
+import org.apache.http.Header;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.Collections;
+import java.util.List;
+
+public class HttpClientJiraIssues extends HttpClient {
+  private final ConfigurationService configurationService;
+  private Logger logger = LoggerFactory.getLogger(HttpClientJiraIssues.class);
+
+  private ObjectMapper mapper = new ObjectMapper();
+
+  public HttpClientJiraIssues(ConfigurationService configurationService) {
+    this.configurationService = configurationService;
+  }
+
+  private CloseableHttpClient getHttpClient() {
+    final RequestConfig requestConfig = RequestConfig.copy(RequestConfig.DEFAULT)
+      .setConnectionRequestTimeout(5000)
+      .setConnectTimeout(5000)
+      .setSocketTimeout(5000)
+      .build();
+    return HttpClients.custom().setDefaultCredentialsProvider(this.configurationService.getJiraCredentials()).setDefaultRequestConfig(requestConfig).build();
+  }
+
+  @Override
+  public String getHttpContent(String url, List<Header> headers) {
+    url = this.configurationService.getJiraURL() + url;
+    return super.getHttpContent(url, headers);
+  }
+
+  public JiraIssues getIssues(String pluginName) throws IOException {
+    return getIssues(pluginName, 0);
+  }
+
+  public JiraIssues getIssues(String pluginName, int startAt) throws IOException {
+    int maxResults = 100;
+    JiraIssues jiraIssues = new JiraIssues();
+
+    String query = URLEncoder.encode("project=JENKINS AND status in (Open, \"In Progress\", Reopened) AND component=" + pluginName + "-plugin", "UTF-8");
```

Comment:
You may need to check if it already ends with plugin, docker-plugin, maven-plugin and simple-theme-plugin all end with that already

Reply:
🤞 there's no tests.